backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchLinkImpl.java

```diff
@@ -111,12 +114,26 @@ void onStart(ConfigurationPropertySource propertySource) {
 			);
 			clientFactoryHolder.close(); // We won't need it anymore
 
-			elasticsearchVersion = ElasticsearchClientUtils.getElasticsearchVersion( clientImplementor );
-			if ( configuredVersionOptional.isPresent() ) {
+			if ( versionCheckEnabled ) {
+				elasticsearchVersion = ElasticsearchClientUtils.getElasticsearchVersion( clientImplementor );
+				if ( configuredVersionOptional.isPresent() ) {
+					ElasticsearchVersion configuredVersion = configuredVersionOptional.get();
+					if ( !configuredVersion.matches( elasticsearchVersion ) ) {
+						throw log.unexpectedElasticsearchVersion( configuredVersion, elasticsearchVersion );
+					}
+				}
+			}
+			else if ( configuredVersionOptional.isPresent() ) {
 				ElasticsearchVersion configuredVersion = configuredVersionOptional.get();
-				if ( !configuredVersion.matches( elasticsearchVersion ) ) {
-					throw log.unexpectedElasticsearchVersion( configuredVersion, elasticsearchVersion );
+				if ( configuredVersion.getMinor().isPresent() && configuredVersion.getMicro().isPresent() ) {
```

Comment:
Ideally this (and the case "no version is configured") should be checked earlier, in `ElasticsearchBackendFactory`. This is especially important for Quarkus, where the configuration is checked at compile-time.

Also, I think you can relax the constraint and allow the micro to be missing: the Elasticsearch team would have to mess up big time for us to have to behave differently based on the micro.

Reply:
thanks for the tips, i will do that in few minutes